### PR TITLE
Add `--build` and `--execute` options to the setup script

### PR DESCRIPTION
**Pull request by nikmolnar**
This PR adds a `--build` option, which will automatically build the Xcode project, and an `--execute` option, which will launch the app in the iOS simulator. Additional `--os` and `--device` options allow the user to build/run the app on different OS/device combinations.

Currently, these options are ignored for all targets other than iOS. ~~I plan to add an implementation for macOS next.~~ (I see this isn't necessary for macOS)

### CODE REVIEW

**Comment by phildini** [comment]
This is flipping amazing. I have a couple comments incoming.

File: briefcase/app.py [hunk 1]
```diff
@@ -52,6 +52,14 @@ class app(Command):
          'URL for the support package to use'),
         ('download-dir=', None,
          "Directory where the project support packages will be cached"),
+        ('build', None,
+         "Build the project after generating"),
+        ('execute', None,
+         "Run the application after building"),
+        ('os-version=', None,
+         "For iOS, the OS version to run (e.g., iOS 10.2)"),
```
Comment (phildini) [inline briefcase/app.py]: Let's not make the docstring iOS-specific. What if we wanted to target a specific android version some day?

Maybe make the docstring something like "Sets the OS version. Currently only iOS supported."

File: briefcase/app.py [hunk 2]
```diff
@@ -52,6 +52,14 @@ class app(Command):
...
+        ('device=', None,
+         "For iOS, the device to run (e.g., iPhone 7 Plus)")
```
Comment (phildini) [inline briefcase/app.py]: See above, re: device. Let's generalize, and leave ourselves open for the future.

File: briefcase/app.py [hunk 3]
```diff
@@ -70,6 +78,10 @@ def initialize_options(self):
         self.version_code = None
         self.guid = None
         self.secret_key = None
+        self.build = False
+        self.execute = False
+        self.os_version = 'iOS 10.2'
```
Comment (phildini) [inline briefcase/app.py]: Related to the above, let's set these props to `None` or something in the general `app.py`, and set the specific values in ios.py

File: briefcase/app.py (hunk 1 above)
Comment (nikmolnar) [inline briefcase/app.py]: Agreed.

File: briefcase/app.py (hunk 3 above)
Comment (nikmolnar) [inline briefcase/app.py]: This is an excellent point; will do.

**Pull request merged**